File: internals/services/html_controller.go
package services

import (
	"fr/diginamic/go-cap/internals/types"
	"fr/diginamic/go-cap/internals/utils"
	"os"
)

func WriteHtmlController(name string) {

    checkDirectoryExists[HTML_CONTROLLER]()

    snakeName := utils.SnakeCase(name)

	controller := types.HtmlController{
		NameSpace: utils.ConvertPathToNameSpace(CONF.JsonController),
		ClassName: name,
        Route: utils.ConvertNameToRoute(name),
        RouteName: utils.ConvertNameToRouteName(name),
        TemplateName: snakeName,
	}

	tmplBytes := getTmplBytes(name, "HtmlController.php", controller)
	path := SRC + CONF.HtmlController + name + "Controller.php"
    templatePath := CONF.TemplateDir + CONF.Page + snakeName
	createFile(tmplBytes, path)
    os.Mkdir(templatePath, 0777)
    createFile(nil, templatePath + "/index.html.twig")
}

File: resources/HtmlController.php
<?php

namespace {{.NameSpace}};

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Attribute\Route;

#[Route('{{.Route}}')]
class {{.ClassName}}Controller extends AbstractController 
{
    #[Route('', methods:['GET'], name: '{{.RouteName}}')]
    public function index(): Response
    {
        return $this->render('{{.TemplateName}}', [
        ]);
    }
}

